ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/services/commands/tracker/CommandTracker.scala

```diff
@@ -197,33 +209,38 @@ private[commands] class CommandTracker[Context](
 
       import CommandTracker.nonTerminalCodes
 
-      private def handleSubmitResponse(submitResponse: Ctx[(Context, String), Try[Empty]]) = {
-        val Ctx((_, commandId), value, _) = submitResponse
+      private def handleSubmitResponse(
+          submitResponse: Ctx[(Context, TrackedCommandKey), Try[Empty]]
+      ): immutable.Iterable[Ctx[Context, Either[CompletionFailure, CompletionSuccess]]] = {
```

Comment:
Maybe a `type SingleCompletionFlowOutput = Ctx[Context, Either[CompletionFailure, CompletionSuccess]]` and other similar typedefs can improve readability